src/init.cpp

```diff
@@ -946,6 +947,18 @@ bool AppInitParameterInteraction()
         }
     }
 
+    // Signal NODE_COMPACT_FILTERS if peercfilters and required index are both enabled.
+    if (gArgs.GetBoolArg("-peercfilters", DEFAULT_PEERCFILTERS)) {
```

Comment:
1b7fbb5

IMO you can set -blockfilterindex by transivity if -peercfilters is already set, just mention it in the helper.